### DISCUSSION TITLE
Using two-keys combinations to switch layers

### DISCUSSION THREAD

**Discussion by sukhe** [kmonad/kmonad#553] [Q&A] [answered]
I have my own keyboard layouts made with AutoHotkey. Different layers are available when pressed  «Space», «Space+Shift», «Space+Ctrl», «Space+Alt» and «Space+Win». That is, to switch layers, I use a combination of two keys. 

I can't figure out how to implement such a switch using KMonad. The tutorial only shows an example of switching with one key. Can anyone help?

**Comment by david-janssen** [comment, marked as answer]
I am not entirely sure what you are describing because of ambiguity in your language, but if I understand correctly, then this is something you could do:

```
(defsrc spc lsft ...)

(defalias
  toS (layer-toggle space)
  toSS (layer-toggle space-shift))

(deflayer base @toS ....)
(deflayer space _ @toSS ...)
(deflayer space-shift ...)
```

So, I elided a lot of stuff there, but in description:
- In your base-layer you have a button that puts you into layer-1 (space)
- In your second-layer you have a button that puts you into layer-2 (space-shift)

So you can get to 1 layer with 1 button, and the other layer with another button.

Does this make sense and address your needs?

**Reply by sukhe** [reply]
Thanks for your reply, David. 

You showed me the right way and I got what I wanted.
Here is a configuration to switch layers using a space, as well as a space with modifiers:

(defalias
ts (tap-hold-next-release 200 space (layer-toggle space))
tss (layer-toggle space-shift)
tsa (layer-toggle space-alt)
tsc (layer-toggle space-ctrl)
tsw (layer-toggle space-win)
)

(deflayer base
...
lctl lmet lalt @tS ralt rmet cmp rctl
)

(deflayer space
...
@tSS z x ...
@tSC @tSW @tSA spc ralt rmet cmp rctl
)

(deflayer space-shift ... )

(deflayer space-alt ... )

(deflayer space-ctrl ... )

(deflayer space-win ... )
